### PR TITLE
Add Langusta de Langton demo

### PR DESCRIPTION
**Pull request by kriskowal**
This is a demonstration of Langton’s Ant https://en.wikipedia.org/wiki/Langton%27s_ant

With one lobster, this is a faithful representation of Langton’s Ant, implemented as a cellular automaton instead of a single-agent automaton.  However, there are additional rules that dictate the behavior of collisions.  Each lobster has a unique priority.  Each lobster makes a bid to follow the course set out for them by the langton rule number beneath them.  For each tile, the neighboring lobster with the highest priority that has bid to move onto that tile advances.  All lobsters that fail their bid turn in place instead.

### CODE REVIEW

File: samples/demos/langtons.lobster [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+// langton rule number beneath them.
+// For each tile, the neighboring lobster with the highest priority that has
+// bid to move onto that tile advances.
+// All lobsters that fail their bid turn in place instead.
+
+import gui
+import texture
+
+def pick(list):
+    return list[rnd(length(list))]
```
Comment (kriskowal) [inline samples/demos/langtons.lobster]: To be removed after #67.

File: samples/demos/langtons.lobster [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+// to the next.
+def transit(len, dst, src, transitions):
+    for(len) i: dst[i] = src[transitions[i]]
+
+// move copies one array to another, given the length of both.
+// This is necessary at the beginning of each tick, to overwrite columns of
+// data from the previous generation.
+def move(len, dst, src):
+    for(len) i:
+        dst[i] = src[i]
```
Comment (kriskowal) [inline samples/demos/langtons.lobster]: Would be neat to have a memcopy builtin for this case. Maybe overload copy, for the no-allocate case. I assume `min(length(src), length(dst))` would be cheap, but I have that figure laying around in this case.

Comment (aardappel) [reply samples/demos/langtons.lobster]: Yup that be nice to have

**Comment by aardappel** [comment]
Thanks!

**Pull request merged**